app/controllers/users_controller.rb

```diff
@@ -0,0 +1,76 @@
+class UsersController < ApplicationController
+	before_action :authenticate_user!
+	before_action :set_user, only: [:show, :update, :destroy, :update_password]
+
+	def index
+		@users = current_user.admin? ? User.all : current_user
+    render json: @users
+  end
+
+  def show
+  	if (@user == current_user) && !(current_user.admin?)
+  		render json: @user
+  	elsif !(@user == current_user) && !(current_user.admin?)
+  		render json: {error: "You are not allowed to view the profile."}, status: 400
+  	else
+  		render json: @user
+  	end
+  end
+
+  def update
+  	if (@user == current_user) && !(current_user.admin?)
+  		@user.update(user_params)
+  		render json: {message: "Successfully updated the profile."}, status: 200
+  	elsif !(@user == current_user) && !(current_user.admin?)
+  		render json: {error: "You are not allowed to update the profile."}, status: 400
+  	else
+  		@user.update(user_params)
+  		render json: {message: "Successfully updated the profile."}, status: 200
+  	end
+  end
+
+  def destroy
+		if (@user == current_user) && !(current_user.admin?)
+  		@user.destroy
+  		render json: {message: "Successfully deleted the profile."}, status: 200
+  	elsif !(@user == current_user) && !(current_user.admin?)
+  		render json: {error: "You are not allowed to delete the profile."}, status: 400
+  	else
+  		@user.destroy
+  		render json: {message: "Successfully deleted the profile."}, status: 200
+  	end
+  end
+
+	def update_password
+		if @user.valid_password?(params[:user][:current_password])
+			if @user.update(update_password_params)
+				bypass_sign_in(@user)
+				render json: {message: "Successfully changed the password."}, status: 200
+			else
+				render json: {error: "Minimum 6 chars required/incorrect confirm password)."}, status: 400
```

Comment:
![image](https://user-images.githubusercontent.com/79498012/112139537-e0869a00-8bf8-11eb-817d-4978d2594ffa.png)
Validation is giving message like this. So that I have written custom message for the api response.